### ISSUE TITLE
Add the ability to pin typography settings

### ISSUE THREAD

**Issue by ariel-maidana** [WordPress/gutenberg#45283]
## What problem does this address?
For some users, it is frustrating to manually open typography settings for each block (like Line Height, Letter Spacing, Decoration, etc.): 
![Screen Capture on 2022-10-25 at 11-59-36](https://user-images.githubusercontent.com/4965723/197809487-527712c0-86e0-4e41-8413-b973abcb4c8f.gif)

## What is your proposed solution?
Add an option in the Editor Settings to pin or leave certain settings open by default in the Block Settings.

**Comment by Psmeg** [comment]
This really needs to be implemented in some form—whether that is through a simple toggle switch for basic/advanced or, at the very least, saving the setting changes to a config or settings file for consistency across the editor.

**Comment by darko018** [comment]
This need to be implement as soon as possible

**Comment by festive-day** [comment]
Please implement via settings or switch

**Comment by afmarchetti** [comment]
This need to be implemented

**Comment by maltmann-muc** [comment]
I fully agree with this feature request. 
Also applies to the other setting sections like dimensions.

**Comment by culturerich** [comment]
Agree with this, but it should apply to all design options (not just Typography), see #55994. 

Plus state persistence across blocks in #55836.

**Comment by Marksson** [comment]
Here's my upvote for this! I totally agree.. and ideally, this kind of improvements are implemented in any place where it makes sense.

**Comment by ahmedzeidan** [comment]
I agree with @Psmeg. It would be nice to have this feature to improve the workflow within the editor.

**Comment by gravnetic** [comment]
The order and pinning of modules is a very good idea! Another approach might be to remember my last state and apply to future editor refresh/load similiar to any design app that exists.

**Comment by ohiawins** [comment]
1UP on this issue. I feel like the setting would be best placed in the Preferences >> Blocks panel which can be accessed anywhere by clicking the vertical elipsis icon in the top menu. 

Uploading pin-type.mov…

<img width="841" alt="Screenshot 2024-07-25 at 11 11 24" src="https://github.com/user-attachments/assets/da934ae9-7d91-426b-9e2b-d3bf7b4c3e39">

**Comment by JakJakson** [comment]
Yep @ariel-maidana  and @Psmeg  - 1 zillion percent this should be implemented ASAP - can't image how many hours over the the course of a year this would save me!

**Comment by martin8032** [comment]
+1

**Comment by mattreport** [comment]
+1 For being able to expose/save these more detailed options in the block panel. I prefer @ohiawins's approach to the issue, having a global setting in the `Preferences > Block panel` window is a sound approach.

**Comment by Psmeg** [comment]
> The order and pinning of modules is a very good idea! Another approach might be to remember my last state and apply to future editor refresh/load similiar to any design app that exists.

I'd love to see both implemented so that new users can set their preferences on first editing content, and anyone using a blueprint could set the preferences and have that globally assigned and available during the deployment phase of the blueprint site.

**Comment by bacoords** [comment]
To see this gain traction, I recommend focusing on a very narrow scope and not adding additional ideas to the thread, even though we'd all agree that "persistence" of settings/panels is a broader issue.

From what I recall, the broader idea of [tab persistence](https://github.com/WordPress/gutenberg/pull/55836) got held up due to accessibility concerns and some upstream changes that needed to be addressed. 

Personally, I also think panel persistence also is [a huge UX issue](https://github.com/WordPress/gutenberg/issues/59789) that makes it worth pursuing a solution, but these should be all a _separate_ conversations on their respective issues so that it doesn't limit the ability for this specific feature to be merged. 

The next steps here would be defining and agreeing on scope- for example:

1. When I open a typography setting, does it stay open just for this single block, for all blocks that are the same type (all paragraphs, for example) or all blocks that have typography settings at all?
2. Should this be available for any "tools" UI in general so that it also applies outside of typography, say to colors, background image, etc?
3. Are we in agreement that this should be an "opt-in" feature that would require a new hidden setting in the preferences panel? Or should it be opt-out? What would that setting be called?

Then once that's done, designing a mockup of what that setting would look like in the Preferences panel.

**Comment by JakJakson** [comment]
Recon it should be "opt-out" @bacoords - else newbie users will get lost - OR- there's a toggle on each setting that clearly says "show all settings" and stays on until its toggled off?

And yeah, any panel that has user editable settings should have he option to "show all"

**Comment by PouicPouic-MCD** [comment]
It should be possible to choose to have some options open by default when loading or editing a page, not only Typography, but all options, Advanced Options etc. Or, as said above, to remember last state.
